mt-kahypar/partition/refinement/deterministic/deterministic_label_propagation.cpp

```diff
@@ -28,27 +28,30 @@
 
 #include "mt-kahypar/definitions.h"
 #include "mt-kahypar/partition/metrics.h"
-#include "mt-kahypar/partition/refinement/gains/km1/km1_attributed_gains.h"
 #include "mt-kahypar/parallel/chunking.h"
 #include "mt-kahypar/parallel/parallel_counting_sort.h"
 #include "mt-kahypar/utils/cast.h"
+#include "mt-kahypar/partition/refinement/gains/gain_definitions.h"
 
 #include <tbb/parallel_sort.h>
 #include <tbb/parallel_reduce.h>
 
 namespace mt_kahypar {
 
-  template<typename TypeTraits>
-  bool DeterministicLabelPropagationRefiner<TypeTraits>::refineImpl(
+
+  template<typename TypeTraits, typename GainTypes>
+  bool DeterministicLabelPropagationRefiner<TypeTraits, GainTypes>::refineImpl(
           mt_kahypar_partitioned_hypergraph_t& hypergraph,
           const vec<HypernodeID>&,
           Metrics& best_metrics,
           const double) {
     PartitionedHypergraph& phg = utils::cast<PartitionedHypergraph>(hypergraph);
     Gain overall_improvement = 0;
     constexpr size_t num_buckets = utils::ParallelPermutation<HypernodeID>::num_buckets;
-    const size_t num_sub_rounds = context.refinement.deterministic_refinement.num_sub_rounds_sync_lp;
-    const size_t num_buckets_per_sub_round = parallel::chunking::idiv_ceil(num_buckets, num_sub_rounds);
+    size_t num_sub_rounds = context.refinement.deterministic_refinement.num_sub_rounds_sync_lp;
+
+    using GainComputation = typename GainTypes::GainComputation;
```

Comment:
define this at class scope

Reply:
Changed to definition at class scope. Btw, a nice effect of doing this locally is that we're not exposing the gain_definitions.h in the header. Slimmer headers lead to faster compile times.